src/ReportManager/SchemaFormContext/SchemaForm/index.js

```diff
@@ -0,0 +1,70 @@
+import React, { forwardRef, useState } from 'react';
+
+import SchemaFormContextProvider from './SchemaFormContext';
+import Section from './fields/Section';
+
+import styles from './styles.module.scss';
+import { FORM_FIELDS_TYPES } from '../../constants';
+import { TextFieldValidators } from './fields/Text';
+import { getSchemaFieldUIType, isFieldActive } from './SchemaFormContext/utils';
+
+const FormFieldValidators = {
+  [FORM_FIELDS_TYPES.TEXT]: TextFieldValidators
+};
+
+const SchemaForm = ({ schema, formData = {}, onFormSubmit, className, onChange }, ref) => {
+
+  const [formErrors, setFormErrors] = useState({});
+
+  const handleOnSubmit = (e) => {
+    e.preventDefault();
+
+    const formErr = { ...formErrors };
+
+    for (const [fieldName, fieldValue] of Object.entries(formData)) {
+
+      if (isFieldActive(schema, fieldName)) {
+        const fieldUIType = getSchemaFieldUIType(schema, fieldName);
+        const validators = FormFieldValidators[fieldUIType];
+
+        for (const [errorType, validatorFn] of Object.entries(validators)) {
+          const hasError = validatorFn(fieldValue, fieldName, schema);
+          if (hasError){
+            formErr[fieldName] = errorType;
+            break; // breaking the loop for the current field to avoid handling multiple errors per field on screen
+          } else {
+            delete formErr[fieldName];
```

Comment:
What was the purpose of creating a `const formErr = { ...formErrors };` if anyway we are going to add manually the errors that are still there and delete the ones that are not? Feels like we could start this formErrors object from scratch.